File: fabric/src/main/java/juuxel/adorn/platform/fabric/EntityBridgeImpl.java
package juuxel.adorn.platform.fabric;

import juuxel.adorn.entity.SeatEntity;
import net.fabricmc.fabric.api.object.builder.v1.entity.FabricEntityTypeBuilder;
import net.minecraft.entity.EntityDimensions;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.SpawnGroup;

public final class EntityBridgeImpl {
    public static EntityType<SeatEntity> createSeatType() {
        return FabricEntityTypeBuilder.create(SpawnGroup.MISC, SeatEntity::new)
            .dimensions(EntityDimensions.fixed(0f, 0f))
            .disableSaving()
            .build();
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/platform/fabric/EntityBridgeImpl.kt
package juuxel.adorn.platform.fabric

import juuxel.adorn.entity.SeatEntity
import juuxel.adorn.platform.EntityBridge
import net.fabricmc.fabric.api.`object`.builder.v1.entity.FabricEntityTypeBuilder
import net.minecraft.entity.EntityDimensions
import net.minecraft.entity.EntityType
import net.minecraft.entity.SpawnGroup

object EntityBridgeImpl : EntityBridge {
    override fun createSeatType(): EntityType<SeatEntity> =
        FabricEntityTypeBuilder.create(SpawnGroup.MISC) { type, world -> SeatEntity(type, world) }
            .dimensions(EntityDimensions.fixed(0f, 0f))
            .build()
}
